File: test/formats/gfx/prs_converter_test.rb
require 'lib/formats/gfx/prs_converter'
require 'lib/virtual_file'
require 'test/test_helper'

# Unit tests for PrsConverter
class PrsConverterTest < Test::Unit::TestCase
  def test_decoding
    input_data = TestHelper.get_test_file('prs/BMIK_A16')
    expected_data = TestHelper.get_test_file('prs/BMIK_A16-out.png')

    file = VirtualFile.new(nil, input_data)
    PrsConverter.decode!(file, {})
    actual_data = file.data

    TestHelper.compare_image(expected_data, actual_data)
  end
end

File: tests/formats/gfx/prs_converter_test.cc
#include "formats/gfx/prs_converter.h"
#include "test_support/converter_support.h"

void test_prs_decoding()
{
    PrsConverter converter;
    assert_decoded_image(
        converter,
        "tests/test_files/gfx/prs/BMIK_A16",
        "tests/test_files/gfx/prs/BMIK_A16-out.png");
}

int main(void)
{
    test_prs_decoding();
    return 0;
}
